crates/biome_js_analyze/src/lint/nursery/use_jsx_curly_brace_convention.rs

```diff
@@ -0,0 +1,339 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsxAttributeValue, AnyJsxChild, JsSyntaxKind,
+    JsSyntaxToken, JsxAttributeInitializerClause, JsxExpressionAttributeValue, T,
+};
+use biome_rowan::{declare_node_union, AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// This rule allows you to enforce curly braces or disallow unnecessary curly braces in JSX props and/or children.
+    ///
+    /// For situations where JSX expressions are unnecessary, please refer to [the React doc](https://facebook.github.io/react/docs/jsx-in-depth.html) and [this page about JSX gotchas](https://github.com/facebook/react/blob/v15.4.0-rc.3/docs/docs/02.3-jsx-gotchas.md#html-entities).
+    ///
+    /// By default, this rule will check for and warn about unnecessary curly braces in both JSX props and children.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <Foo>{'Hello world'}</Foo>;
+    ///
+    /// <Foo foo={'bar'} />;
+    ///
+    /// <Foo foo=<Bar /> />;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <Foo>Hello world</Foo>;
+    /// <Foo foo="bar" />;
+    /// <Foo foo={5} />;
+    /// <Foo foo={<Bar />} />;
+    /// ```
+    ///
+    pub UseJsxCurlyBraceConvention {
+        version: "next",
+        name: "useJsxCurlyBraceConvention",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintReact("jsx-curly-brace-presence")],
+        source_kind: RuleSourceKind::Inspired,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+declare_node_union! {
+    pub UseJsxCurlyBraceConventionQuery = JsxAttributeInitializerClause | AnyJsxChild
```

Comment:
When declaring a union of nodes, we always use the convention `Any*`. So, in this case, I would call this node `AnyJsxQueryNode`, or something that starts with `Any*`